playbooks/common/openshift-cluster/upgrades/post_control_plane.yml

```diff
@@ -5,10 +5,14 @@
 - name: Upgrade default router and default registry
   hosts: oo_first_master
   vars:
-    registry_image: "{{  openshift.master.registry_url | replace( '${component}', 'docker-registry' )  |
-                         replace ( '${version}', openshift_image_tag ) }}"
-    router_image: "{{ openshift.master.registry_url | replace( '${component}', 'haproxy-router' ) |
-                      replace ( '${version}', openshift_image_tag ) }}"
+    registry_image: "{{ openshift_hosted_registry_registryurl |
+                        default(openshift.master.registry_url) |
+                        replace('${component}', 'docker-registry') |
+                        replace('${version}', openshift_image_tag) }}"
+    router_image: "{{ openshift_hosted_router_registryurl |
+                      default(openshift.master.registry_url) |
+                      replace('${component}', 'haproxy-router') |
+                      replace('${version}', openshift_image_tag) }}"
     registry_console_image: "{{ openshift.master.registry_url | regex_replace ( '(origin|ose)-\\${component}', 'registry-console') |
```

Comment:
This should presumably use the same logic as `registry_image`.

Reply:
Not quite, the console image is handled differently in the playbooks (https://github.com/openshift/openshift-ansible/blob/196891f1a71fa0f0c74404af773d53ede517025a/inventory/byo/hosts.example#L402-L407) and deployed with a template (in the cockpit-ui role).

So the correct way would be to actually look at the current deployment and figure it out from there.